pymc/tests/test_distributions_timeseries.py

```diff
@@ -142,6 +143,28 @@ def test_gaussian_random_walk_init_dist_logp(self, init):
             pm.logp(init, 0).eval() + scipy.stats.norm.logpdf(0),
         )
 
+    @pytest.mark.parametrize(
+        "mu, sigma, init, steps, size, expected",
+        [
+            (0, 1, Normal.dist(1), 10, None, np.ones((11,))),
+            (1, 1, Normal.dist(0), 10, (2,), np.full((2, 11), np.arange(11))),
+            (1, 1, Normal.dist([0, 1]), 10, None, np.vstack((np.arange(11), np.arange(11) + 1))),
+            (0, [1, 1], Normal.dist(0), 10, None, np.zeros((2, 11))),
+            (
+                [1, -1],
+                1,
+                Normal.dist(0),
+                10,
+                (4, 2),
+                np.full((4, 2, 11), np.vstack((np.arange(11), -np.arange(11)))),
+            ),
+        ],
+    )
+    def test_moment(self, mu, sigma, init, steps, size, expected):
+        with Model() as model:
+            GaussianRandomWalk("x", mu=mu, sigma=sigma, init=init, steps=steps, size=size)
+        assert_moment_is_expected(model, expected)
```

Comment:
These tests cover the changes to the logp and random methods because they invoke both